### PR TITLE
[MIG] stock_partner_delivery_window: Migration to 14.0

### PR DESCRIPTION
**Pull request by dzungtran89**
Module dependency:

- [x] `base_time_window` https://github.com/OCA/server-tools/pull/1988
- [x] `partner_tz`: migrated in https://github.com/OCA/partner-contact/pull/1043

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+odoo14-addon-partner-tz @ git+https://github.com/trobz/partner-contact@14.0-mig-partner_tz#subdirectory=setup/partner_tz
```
Comment (lmignon) [inline test-requirements.txt]: @trobz can you add into your PR description than this one depends on others PR?

Comment (dzungtran89) [reply test-requirements.txt]: @lmignon ok, I updated the PR description

**Comment by Stephanisk** [comment]
Hi,

Does someone have any news on the migration of this module to V14. I am really lost without this module.

Thx

File: oca_dependencies.txt [hunk 2]
```diff
@@ -1 +1,3 @@
 # See https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#oca_dependencies-txt
```
Comment (simahawk) [inline oca_dependencies.txt]: you can trash this file: [MQT=PIP](https://github.com/OCA/stock-logistics-workflow/blob/14.0/.travis.yml#L30)

File: test-requirements.txt (hunk 1 above)
Comment (simahawk) [reply test-requirements.txt]: it depends on this as well https://github.com/OCA/server-tools/pull/1988 (I've added it to description)

**Review by simahawk — CHANGES_REQUESTED** [review]
LG overall. Would you mind cherry-picking these changes?

- [ ] https://github.com/OCA/stock-logistics-workflow/pull/822
- [ ] https://github.com/OCA/stock-logistics-workflow/pull/824
- [ ] https://github.com/OCA/stock-logistics-workflow/pull/830

**Comment by simahawk** [comment]
> Hi,
> 
> Does someone have any news on the migration of this module to V14. I am really lost without this module.
> 
> Thx

You can help w/ functional and / or technical review to speed it up (same for the PRs that are dependencies for this one).
Meanwhile, you can use it from here. I suggest you give a try to https://pypi.org/project/git-aggregator/ to collect pending PRs alltogether.

**Comment by Stephanisk** [comment]
I would love to help, just don't know how I could.

You said, meanwhile you can use it from here. What do you mean by this? I went to the aggregator website but I don't understand at all what I could do there/with that.

Sorry, I am learning more everyday, but this whole procedure on GitHub still gets me rattled when it comes to Odoo, it's complex.

**Comment by simahawk** [comment]
You can also install modules directly from PRs. Add this to your requirements.txt:

```
odoo14-addon-partner_tz @ git+https://github.com/OCA/partner-contact@refs/pull/1043/head#subdirectory=setup/partner_tz
odoo14-addon-base_time_window @ git+https://github.com/OCA/server-tools@pull/1988/head#subdirectory=setup/base_time_window
odoo14-addon-stock_partner_delivery_window @ git+https://github.com/OCA/stock-logistics-workflow@refs/pull/775/head#subdirectory=setup/stock_partner_delivery_window
```

and run `pip install -r requirements.txt`. Then you can install modules as usual w/ `-i`.

**Comment by Stephanisk** [comment]
Hi,

I am trying to wrap my head around it.


So I can install from updated pull requests no matter if the module is made
for v14?

For that I make a file requirements.txt with the content you placed and I
run a pip install.

I get lost now you say I can install modules as usual?? Where would I find
the module partner delivery window? Inside Odoo apps?

This is a production environment, if it would cause problems how will I
uninstall these modules?

Thanks for all your help so far. Send me your address please so I can send
you a thank you package .

Regards,
Stephane

**Comment by simahawk** [comment]
Once you run pip install as mentioned above you'll have the modules available for installing. 

I gave you this option because it avoids you the need of understanding and using git-aggregator.

If you are not familiar w/ python installations nor w/ git-aggregator nor w/ how to checkout PRs locally I'd recommend you ask for help to a professional (either for training of for technical work).

> This is a production environment, if it would cause problems how will I
uninstall these modules?

Never do this on prod envs if you don't know what you are doing and if you don't know how to rollback things.

You should try this locally 1st or on a test instance.

File: oca_dependencies.txt (hunk 2 above)
Comment (dzungtran89) [reply oca_dependencies.txt]: hi @simahawk , thx for your review. These changes have been cherry-picked to this MR

File: stock_partner_delivery_window/models/res_partner.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    _inherit = "res.partner"
+
+    delivery_time_preference = fields.Selection(
+        [
+            ("anytime", "Any time"),
+            ("time_windows", "Fixed time windows"),
+            ("workdays", "Weekdays (Monday to Friday)"),
+        ],
+        string="Delivery time schedule preference",
+        default="workdays",
```
Comment (simahawk) [inline stock_partner_delivery_window/models/res_partner.py]: @grindtildeath @mmequignon the default here should be changed?

**Review by simahawk — COMMENTED** [review]
@grindtildeath @mmequignon please review

File: stock_partner_delivery_window/models/res_partner.py (hunk 3 above)
Comment (grindtildeath) [reply stock_partner_delivery_window/models/res_partner.py]: Yes, please cherry-pick https://github.com/OCA/stock-logistics-workflow/pull/862 :pray:

Comment (hailangvn) [reply stock_partner_delivery_window/models/res_partner.py]: Thanks @simahawk , @grindtildeath for your comments. We are working on this.

Comment (hailangvn) [reply stock_partner_delivery_window/models/res_partner.py]: @simahawk, @grindtildeath, The PR has been updated.

File: test-requirements.txt (hunk 1 above)
Comment (simahawk) [inline test-requirements.txt]: you don't need these anymore

**Review by simahawk — APPROVED** [review]
LGTM but test requirements

**Comment by Stephanisk** [comment]
Hi, I see Travis made test that didn't conflict. does this mean that the migration will soon be done.

Do you need any help in translations? I would be glad to help out

**Comment by simahawk** [comment]
@grindtildeath and/or @mmequignon can you review?

**Review by sebalix — APPROVED** [review]

**Pull request merged**